doc/csv/recipes.rdoc

```diff
@@ -0,0 +1,330 @@
+
+Instance method CSV#each can read a source \String one row at a time:
+  CSV.new(string, headers: true).each do |row|
+    p row
+  end
+Ouput:
+  #<CSV::Row "Name":"foo" "Value":"0">
+  #<CSV::Row "Name":"bar" "Value":"1">
+  #<CSV::Row "Name":"baz" "Value":"2">#
+
+==== Parse from \File Without Headers
+
+Instance method CSV#read can read a file all at once:
+  string = "foo,0\nbar,1\nbaz,2\n"
+  CSV.new(string).read # => [["foo", "0"], ["bar", "1"], ["baz", "2"]]
+
+\Class method CSV::foreach can read one row at a time:
+  path = 't.csv'
+  File.write(path, string)
+  CSV.foreach(path) do |row|
+    p row
+  end
+Output:
+  ["foo", "0"]
+  ["bar", "1"]
+  ["baz", "2"]
+
+==== Parse from \File with Headers
+
+Instance method CSV#read can reada file all at once:
+  string = "Name,Value\nfoo,0\nbar,1\nbaz,2\n"
+  path = 't.csv'
+  File.write(path, string)
+  CSV.read(path, headers: true) # => #<CSV::Table mode:col_or_row row_count:4>
+
+\Class method CSV::foreach can read one row at a time:
+  CSV.foreach(path, headers: true) do |row|
+    p row
+  end
+Output:
+  #<CSV::Row "Name":"foo" "Value":"0">
+  #<CSV::Row "Name":"bar" "Value":"1">
+  #<CSV::Row "Name":"baz" "Value":"2">
+
+==== Parse from \IO Stream Without Headers
+
+\Class method CSV::parse can read an \IO stream all at once:
+  string = "foo,0\nbar,1\nbaz,2\n"
+  path = 't.csv'
+  File.write(path, string)
+  File.open(path) do |file|
+    CSV.parse(file)
+  end # => [["foo", "0"], ["bar", "1"], ["baz", "2"]]
+
+\Class method CSV::foreach can read one row at a time:
+  File.open(path) do |file|
+    CSV.foreach(file) do |row|
+      p row
+    end
+  end
+Output:
+  ["foo", "0"]
+  ["bar", "1"]
+  ["baz", "2"]
+
+==== Parse from \IO Stream with Headers
+
+\Class method CSV::parse can read an \IO stream all at once:
+  string = "Name,Value\nfoo,0\nbar,1\nbaz,2\n"
+  path = 't.csv'
+  File.write(path, string)
+  File.open(path, headers: true) do |file|
+    CSV.parse(file)
+  end # => [["Name", "Value"], ["foo", "0"], ["bar", "1"], ["baz", "2"]]
+
+\Class method CSV::foreach can read one row at a time:
+  File.open(path, headers: true) do |file|
+    CSV.foreach(file) do |row|
+      p row
+    end
+  end
+Output:
+  ["Name", "Value"]
+  ["foo", "0"]
+  ["bar", "1"]
+  ["baz", "2"]
+
+=== Generating
+
+==== Generate to \String Without Headers
+
+\Class method CSV::generate can generate to a \String.
+
+This example uses method CSV#<< to append the rows
+that are to be generated:
+  output_string = CSV.generate do |csv|
+    csv << ['Foo', 0]
+    csv << ['Bar', 1]
+    csv << ['Baz', 2]
+  end
+  output_string # => "foo,0\nbar,1\nbaz,2\n"
+
+==== Generate to \String with Headers
+
+\Class method CSV::generate can generate to a \String.
+
+This example uses method CSV#<< to append the rows
+that are to be generated:
+  output_string = CSV.generate('', headers: ['Name', 'Value']) do |csv|
+    csv << csv.headers
```

Comment:
```suggestion
  output_string = CSV.generate('', headers: ['Name', 'Value'], write_headers: true) do |csv|
```

Reply:
Could you confirm this too?

Reply:
Fixed in new push (plus, one more similar).